pkg/registries/registries.go

```diff
@@ -180,3 +182,27 @@ func (ms *mirrorSet) generateConfig() string {
 
 	return result
 }
+
+// GenerateConfigFiles uses cluster resources to generate the containerd mirror registry configuration files
+func GenerateConfigFiles(ctx context.Context, c client.Client) (map[string][]byte, error) {
+	// List IDMS/ITMS resources
+	imageDigestMirrorSetList := &config.ImageDigestMirrorSetList{}
+	if err := c.List(ctx, imageDigestMirrorSetList); err != nil {
+		return nil, fmt.Errorf("error getting IDMS list: %w", err)
+	}
+	imageTagMirrorSetList := &config.ImageTagMirrorSetList{}
+	if err := c.List(ctx, imageTagMirrorSetList); err != nil {
+		return nil, fmt.Errorf("error getting ITMS list: %w", err)
+	}
+
+	registryConf := newRegistryConfig(imageDigestMirrorSetList.Items, imageTagMirrorSetList.Items)
```

Comment:
Please reevaluate this object and how it is being used.

As it is now, this function would be better as `getMirrorsets`